### ISSUE TITLE
TypeError: Cannot read property '_update' of undefined

### ISSUE THREAD

**Issue by sergebesson** [Automattic/mongoose#8155]
**Do you want to request a *feature* or report a *bug*?**
BUG

**What is the current behavior?** 
TypeError: Cannot read property '_update' of undefined

**If the current behavior is a bug, please provide the steps to reproduce.**
Here's a sample of code to reproduce the bug
```javascript
const mongoose = require("mongoose");

const exampleSchema = new mongoose.Schema({
	name: { type: String },
	value: { type: Number },
}, {
	collection: "example",
	strict: "throw",
});
const ExampleModel = mongoose.model("Model", exampleSchema);
const example = new ExampleModel({ name: "example 1", value: 1 });

return mongoose
	.connect('mongodb://localhost:27017/test', {
		useNewUrlParser: true, useUnifiedTopology: true
	})
	.then(() => example.save())
	.then(() => ExampleModel.findOneAndUpdate({
			name: "example 1",
		} , {
			$set: { value: 2 },
		}, {
			runValidators: true,
		})
		.select("value")
	)
	.then((result) => console.log("test OK", result))
	.finally(() => mongoose.disconnect())
	.catch((error) => console.log("ERROR", error.stack));
```

**What is the expected behavior?**
we expected the document to be updated

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
* node : 10.16.0
* mongoose : 5.7.0
* mongoDb: 3.6.12

**Comment by sergebesson** [comment]
the bug does not exist in version 5.6.13

**Comment by luizsantos19922** [comment]
I had the same problem.

**Comment by baublet** [comment]
@kevin-wynn and I just spent all day debugging this. Apparently there an API breaking change in several hooks (in our case, in a `pre` hook) between 5.6 and 5.7. To fix this, *lock your version numbers tightly*. We we locked our to an older version:

```
"mongoose": "^5.2.18",
```

```
"mongoose": "5.2.x",

// OR

"mongoose": "5.2.18",
```

Or, if the above comments are correct about `5.6.13`:

```
"mongoose": "5.6.13",
```

**Comment by baublet** [comment]
And FYI, this PR looks like it might fix this issue: https://github.com/Automattic/mongoose/pull/8151/files

**Comment by vkarpov15** [comment]
This is fixed in [5.7.1](https://github.com/Automattic/mongoose/blob/master/History.md#571--2019-09-13). Please upgrade and the issue should go away. Sorry for the trouble!

**Comment by sergebesson** [comment]
I confirm that version 5.7.1 fix the issue.
thank you very much :-)
